ddtrace/settings/config.py

```diff
@@ -187,6 +196,72 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+ConfigSource = Literal["default", "env", "code"]
+
+
+class _ConfigItem:
+    """Configuration item that tracks the value of a setting, and where it came from."""
+
+    def __init__(self, name, default, envs):
+        self._name = name
+        self._default_value = default
+        self._env_value = None
+        self._code_value = None
+        self._envs = envs
+        for (env_var, parser) in envs:
+            if env_var in os.environ:
+                self._env_value = parser(os.environ[env_var])
+                break
+
+    def set_code(self, value):
+        self._code_value = value
+
+    def value(self):
```

Comment:
Just to confirm my understanding - we're always going to resolve a config value based on this order [user provided override, env var, then default]?

Reply:
correct, if there is a need to have a different precedence then we could subclass `_ConfigItem` and override the value method for that setting